### Alunos
NATHAN CARDOSO LINHARES  2026010155

PEDRO VINICIUS DE ANDRADE QUEIROZ 2026010152     

Adicionando o dataset com base nas instruções do próprio site. As instruções estão disponíveis em: https://archive.ics.uci.edu/dataset/294/combined+cycle+power+plant

In [1]:
from ucimlrepo import fetch_ucirepo 
  
# fetch dataset 
combined_cycle_power_plant = fetch_ucirepo(id=294) 
  
# data (as pandas dataframes) 
X = combined_cycle_power_plant.data.features 
y = combined_cycle_power_plant.data.targets 

# variable information 
display(X) 
display(y) 


,AT,V,AP,RH
0,14.96,41.76,1024.07,73.17
1,25.18,62.96,1020.04,59.08
2,5.11,39.40,1012.16,92.14
3,20.86,57.32,1010.24,76.64
4,10.82,37.50,1009.23,96.62
...,...,...,...,...
9563,16.65,49.69,1014.01,91.00
9564,13.19,39.18,1023.67,66.78
9565,31.32,74.33,1012.92,36.48
9566,24.48,69.45,1013.86,62.39


,PE
0,463.26
1,444.37
2,488.56
3,446.48
4,473.90
...,...
9563,460.03
9564,469.62
9565,429.57
9566,435.74


Após a obtenção dos dados o passo seguinte foi realizar a concatenação das variáveis x, referente as features, e y, que diz respeito aos targets.
A concatenação é armazenada na veriável `df` para representar tipo de dado dataframe.

In [2]:
import pandas as pd

df = pd.concat([X, y], axis=1)

display(df)

,AT,V,AP,RH,PE
0,14.96,41.76,1024.07,73.17,463.26
1,25.18,62.96,1020.04,59.08,444.37
2,5.11,39.40,1012.16,92.14,488.56
3,20.86,57.32,1010.24,76.64,446.48
4,10.82,37.50,1009.23,96.62,473.90
...,...,...,...,...,...
9563,16.65,49.69,1014.01,91.00,460.03
9564,13.19,39.18,1023.67,66.78,469.62
9565,31.32,74.33,1012.92,36.48,429.57
9566,24.48,69.45,1013.86,62.39,435.74


Em seguida, foi calculado a correlação das variáveis AT, V, AP e RH em relação a PE.

In [3]:
corr = df.corr()["PE"].drop("PE")
display(corr)

AT   -0.948128
V    -0.869780
AP    0.518429
RH    0.389794
Name: PE, dtype: float64

O próximo passo foi obter a matriz de correlação para observar a correlação de todas as variáveis do dataset.

In [4]:
display(df.corr())


,AT,V,AP,RH,PE
AT,1.000000,0.844107,-0.507549,-0.542535,-0.948128
V,0.844107,1.000000,-0.413502,-0.312187,-0.869780
AP,-0.507549,-0.413502,1.000000,0.099574,0.518429
RH,-0.542535,-0.312187,0.099574,1.000000,0.389794
PE,-0.948128,-0.869780,0.518429,0.389794,1.000000


Por fim, foi realizado a predição de PE e observado as métricas de desempenhos MSE, MAE e RMSE e r².

In [6]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
import numpy as np

# Divisão treino/teste (80/20)
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

# Modelo
model = LinearRegression()
model.fit(X_train, y_train)

# Previsões
y_pred = model.predict(X_test)

# Métricas
mse = mean_squared_error(y_test, y_pred)
rmse = np.sqrt(mse)
mae = mean_absolute_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

print("MSE:", mse)
print("RMSE:", rmse)
print("MAE:", mae)
print("R²:", r2)

print("\nCoeficientes:")
coef_df = pd.DataFrame({
    "Variável": X.columns,
    "Coeficiente": model.coef_.ravel()
})
display(coef_df)

print("\nIntercepto:", model.intercept_)

MSE: 20.27370599968744
RMSE: 4.502633229532186
MAE: 3.595913178273408
R²: 0.9301046431962188

Coeficientes:


,Variável,Coeficiente
0,AT,-1.985900
1,V,-0.232094
2,AP,0.062200
3,RH,-0.158118



Intercepto: [454.56911459]


## Análise e respostas.

- 1 **Resposta**

Resultado da correlação:

AT   -0.948128

V    -0.869780

AP    0.518429

RH    0.389794

Nota-se que as variáveis AP e RH são as variáveis com maior influência, pelo fato de serem as duas com correlação positiva. Diferente das demais AT e V que apresentam valores negativos.

---

- 2 **Resposta**:

A análise da matriz de correlação entre as variáveis, mostra que a maioria das relações apresenta correlações fracas a moderadas. No entanto observa-se uma forte correlação positiva entre AT e V (0,84), porém, são variáveis que fazem parte do grupo de variáveis dependentes. Sendo uma baixa indicação de multicolinearidade.

---

- 3 **Resposta**:




```bash
# Divisão treino/teste (80/20)
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

# Modelo
model = LinearRegression()
model.fit(X_train, y_train)
```

O trecho de código acima apresenta a divisão sugerida.

---

- 4 **Resposta**

O modelo apresentou um excelente desempenho. O valor de R² igual a 0,93 indica que aproximadamente 93% da variabilidade da produção de energia (PE) é explicada pelas variáveis independentes (AT, V, AP e RH), demonstrando alto poder explicativo.

Além disso, os valores de RMSE (4,50) e MAE (3,59) são relativamente baixos, indicando que os erros de previsão do modelo são pequenos em relação à escala da variável dependente.Portanto, o modelo é adequado para prever a produção de energia, pois apresenta alta capacidade de explicação e baixos erros de previsão.
